### PR TITLE
Fix psalm errors and upgrade strictness

### PR DESCRIPTION
**Pull request by orklah**
Hi,

This PR fix a new batch of issues reported by Psalm or PHPStorm.

It also upgrade the errorLevel from 8 to 7.

### CODE REVIEW

File: lib/Doctrine/ORM/AbstractQuery.php [hunk 1]
```diff
@@ -337,9 +338,11 @@ function (Query\Parameter $parameter) use ($key) : bool {
     /**
      * Sets a collection of query parameters.
      *
-     * @param ArrayCollection|mixed[] $parameters
+     * @param ArrayCollection|Parameter[] $parameters
```
Comment (greg0ire) [inline lib/Doctrine/ORM/AbstractQuery.php:341]: This does not look right… on line 354, `$value` is could be a `Parameter` instance if I look at the right hand part of this phpdoc type, and I don't think `new Parameter($key, $value)` makes sense in that case.

**Comment by greg0ire** [comment]
It's weird that github says you are a first-time contributor to this repository :thinking:

File: lib/Doctrine/ORM/AbstractQuery.php (hunk 1 above)
Comment (orklah) [reply lib/Doctrine/ORM/AbstractQuery.php:341]: You're right. I was confused by `ArrayCollection|mixed[]` that I assumed was a way to express `ArrayCollection<mixed>` and then, I saw that it was assigned to `$this->parameters` who is indeed an `ArrayCollection<Parameter>`. In fact, the param is `ArrayCollection<Parameter>|mixed[]`. I'll fix this

**Comment by orklah** [comment]
> It's weird that github says you are a first-time contributor to this repository 🤔

Maybe it's only counting the main branch?

**Review by greg0ire — APPROVED** [review]

**Review by SenseException — APPROVED** [review]

**Pull request merged**